packages/flow/lib/iterate.d.ts

```diff
@@ -1,5 +1,9 @@
 /** Iterates the given object and returns the value of the given function using each value. */
-export function iterate <T>(
-	iteratable: Iterable<T>,
-	render?: ((value: T) => any)
+
+import { GetAppropriateFunctionBasedOnWhetherOrNotAGeneratorOfAnIterableWithTheForEachMethodIsPassed, } from "./shared";
+
+
+export function iterate<T extends Iterable<unknown>, U extends GetAppropriateFunctionBasedOnWhetherOrNotAGeneratorOfAnIterableWithTheForEachMethodIsPassed<T> >(
```

Comment:
This is the right type for the `iterate` method.  The function that is passed in is done based on the iterable that is passed in. If the iterable has a `forEach()` method then the function will pass in the value, key or index , and the iterable to the callback function If not the value will be passed